src/System.Device.Gpio/System/Device/Gpio/Drivers/LibGpiodDriver.Linux.cs

```diff
@@ -21,29 +22,48 @@ public class LibGpiodDriver : UnixDriver
 
         public LibGpiodDriver()
         {
-            SafeChipIteratorHandle iterator = null;
-            try
-            {
-                iterator = Interop.GetChipIterator();
-                if (iterator == null)
-                {
-                    throw ExceptionHelper.GetIOException(ExceptionResource.NoChipIteratorFound, Marshal.GetLastWin32Error());
-                }
-            }
-            catch (DllNotFoundException)
-            {
-                throw ExceptionHelper.GetPlatformNotSupportedException(ExceptionResource.LibGpiodNotInstalled);
-            }
-            
-            _chip = Interop.GetNextChipFromChipIterator(iterator);
-            if (_chip == null)
-            {
-                throw ExceptionHelper.GetIOException(ExceptionResource.NoChipFound, Marshal.GetLastWin32Error());
-            }
-
-            _pinNumberToSafeLineHandle = new Dictionary<int, SafeLineHandle>(PinCount);
+			// set /dev/gpiochip0 as default if hardware has only one GPIO bank
+			if (Interop.GetNumberOfChips() == 1)
+            	SetDefaultGpioChip(0);
         }
 
+		protected internal override void SetDefaultGpioChip(int devGpioChip)
+		{
+			if (!_gpiochip_seted) 
+			{
+				SafeChipIteratorHandle iterator = null;
+				try
+				{
+					iterator = Interop.GetChipIterator();
+					if (iterator == null)
+					{
+						throw ExceptionHelper.GetIOException(ExceptionResource.NoChipIteratorFound, Marshal.GetLastWin32Error());
+					}
+				}
+				catch (DllNotFoundException)
+				{
+					throw ExceptionHelper.GetPlatformNotSupportedException(ExceptionResource.LibGpiodNotInstalled);
+				}
+
+				// here we are getting the default controller /dev/gpiochipX
+				for (int i = 0; i <= devGpioChip; i++) 
+				{
+					_chip = Interop.GetNextChipFromChipIterator(iterator);
```

Comment:
you can load gpio chips directly by name or chim number https://git.kernel.org/pub/scm/libs/libgpiod/libgpiod.git/tree/include/gpiod.h#n416.
So for me we don't need to iterate by each chip, further seems don't see use of GetNumberOfChips method and related logic, can be just load chip 0 by default if chip number not provided

Reply:
Thanks for the review @buyaa-n 
In b6cb398 I changed the iterator way to gpiod_chip_open_by_number